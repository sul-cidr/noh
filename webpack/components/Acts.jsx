import React, { Component } from "react";
import PropTypes from "prop-types";

class Acts extends Component {
  createActs() {
    const length = this.props.acts.length - 1;
    let left = 0;
    return this.props.acts.map((act, index) => {
      const width = (100 * act.duration) / this.props.duration;
      const transcription = act.transcription ? (
        <span className="transcription">{act.transcription}</span>
      ) : (
        ""
      );
      left += width;
      const actEnd =
        index < length ? (
          <div
            key={`act-end-${act.translation}-${width}`}
            className="act__first-end"
            style={{ left: `${left}%` }}
          />
        ) : (
          ""
        );
      return [
        <div
          key={`act-${act.translation}-${width}`}
          className="act"
          style={{ width: `${width}%` }}
        >
          {transcription}
        </div>,
        actEnd
      ];
    });
  }

  render() {
    const acts = this.createActs();
    return <div className="acts-container">{acts}</div>;
  }
}

Acts.propTypes = {
  acts: PropTypes.arrayOf(
    PropTypes.shape({
      translation: PropTypes.string.isRequired,
      transcription: PropTypes.string.isRequired,
      duration: PropTypes.number.isRequired
    })
  ).isRequired,
  duration: PropTypes.number.isRequired
};

export default Acts;
