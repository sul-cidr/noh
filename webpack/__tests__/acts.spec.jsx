import React from "react";
import { shallow } from "enzyme";
import Acts from "../components/Acts";

describe("<Acts>", () => {
  it("renders as expected by default", () => {
    const acts = [
      { translation: "Musicians", transcription: "", duration: 360 },
      { translation: "First Act", transcription: "Mae ba", duration: 1293 },
      { translation: "Second Act", transcription: "Nochi ba", duration: 2400 },
      { translation: "", transcription: "Oshirabe", duration: 360 }
    ];
    const component = shallow(<Acts acts={acts} duration={4413} />);
    expect(component).toMatchSnapshot();
  });
});
