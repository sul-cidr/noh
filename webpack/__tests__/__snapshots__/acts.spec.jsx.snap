// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Acts> renders as expected by default 1`] = `
<div
  className="acts-container"
>
  <div
    className="act"
    key="act-Musicians-8.157715839564922"
    style={
      Object {
        "width": "8.157715839564922%",
      }
    }
  >
    <span>
      Musicians
    </span>
  </div>
  <div
    className="act__first-end"
    key="act-end-Musicians-8.157715839564922"
    style={
      Object {
        "left": "8.157715839564922%",
      }
    }
  />
  <div
    className="act"
    key="act-First Act-29.29979605710401"
    style={
      Object {
        "width": "29.29979605710401%",
      }
    }
  >
    <span>
      First Act
    </span>
  </div>
  <div
    className="act__first-end"
    key="act-end-First Act-29.29979605710401"
    style={
      Object {
        "left": "37.45751189666893%",
      }
    }
  />
  <div
    className="act"
    key="act-Second Act-54.384772263766145"
    style={
      Object {
        "width": "54.384772263766145%",
      }
    }
  >
    <span>
      Second Act
    </span>
  </div>
  <div
    className="act__first-end"
    key="act-end-Second Act-54.384772263766145"
    style={
      Object {
        "left": "91.84228416043507%",
      }
    }
  />
  <div
    className="act"
    key="act--8.157715839564922"
    style={
      Object {
        "width": "8.157715839564922%",
      }
    }
  />
</div>
`;
